target/linux/ramips/dts/mt7628an_tplink_re200-v2.dts

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "tplink,re200-v2", "mediatek,mt7628an-soc";
+	model = "TP-Link RE200 v2";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600n8";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wps {
+			label = "re200-v2:green:wps";
+			gpios = <&gpio0 5 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi {
+			label = "re200-v2:green:wifi";
+			gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "re200-v2:green:lan";
+			gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power: power {
+			label = "re200-v2:green:power";
+			gpios = <&gpio1 12 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi2g_green {
+			label = "re200-v2:green:wifi2g";
+			gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		wifi2g_red {
+			label = "re200-v2:red:wifi2g";
+			gpios = <&gpio1 11 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi5g_green {
+			label = "re200-v2:green:wifi5g";
+			gpios = <&gpio1 10 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		wifi5g_red {
+			label = "re200-v2:red:wifi5g";
+			gpios = <&gpio1 8 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+
+&state_default {
+	gpio {
+		ralink,group = "p4led_an", "p3led_an", "p2led_an", "p1led_an", "p0led_an", "wled_an", "i2c", "uart0";
+		ralink,function = "gpio";
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+
+			partition@20000 {
+				compatible = "tplink,firmware";
+				label = "firmware";
+				reg = <0x20000 0x7a0000>;
+			};
+
+			partition@7c0000 {
```

Comment:
It's actually unusual to have partition-table etc. as separate partitions here.
If you look at RE305 or RE350 (mt7621), you will see that those partitions are typically just set up as one config partition in DTS (reg = <0x7c0000 0x30000> in your case). I'd do the same here (do not forget to update MAC address locations).

Reply:
Yep, I saw that, and also a comment in the PR for the RE305 regarding the combination. Personally, I prefer separate partitions because it's more consistent with the definitions in tplink-safeloader.c. For consistency reasons within the .dts files, I'll change that - tomorrow, enough for today.

Reply:
There recently was a similar case on the openwrt-devel mailing list, where a device was migrated from ar71xx to ath79 and the same question arose. There, request was also to have one config partition. (Unfortunately I do not find it in the archive ...)